src/program.cpp

```diff
@@ -353,6 +383,10 @@ bool spir_binary::parse_kernel(const std::vector<std::string>& tokens,
         if (!parse_local_arg(arg, tokens, toknum)) {
             return false;
         }
+    } else if (tokens[toknum] == "offset") {
```

Comment:
Another example of a needed change to the descriptor map. We should probably rework the format so that `argKind` comes before `offset` and select what non-descriptor entry is going to be parsed next. Not saying this has to be done now though.

Reply:
Yes, I agree. I have further descriptor map changes coming in clspv though so probably should wait till it stabilizes some more.